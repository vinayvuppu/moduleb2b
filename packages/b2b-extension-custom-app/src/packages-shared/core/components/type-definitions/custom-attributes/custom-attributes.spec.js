import React from 'react';
import { shallow } from 'enzyme';
import AttributeInput from '../../attribute-definitions/attribute-input';
import CustomAttributes from './custom-attributes';

const createTestProps = props => ({
  currencies: ['EUR', 'USD'],
  languages: ['en', 'de'],
  language: 'en',
  expandSettings: {},
  fieldDefinitions: [
    {
      name: 'test',
      type: {
        name: 'String',
      },
    },
    {
      name: 'boolean',
      type: {
        name: 'Boolean',
      },
    },
    {
      name: 'set-dates',
      type: {
        name: 'set',
        elementType: {
          name: 'date',
        },
      },
    },
  ],
  fields: {
    test: 'value',
    boolean: false,
    'set-dates': ['2017-01-27T09:40:38.179Z', '2017-05-27T09:40:38.179Z'],
  },
  updateSettings: jest.fn(),
  handleChange: jest.fn(),
  isDisabled: false,
  ...props,
});
describe('rendering', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<CustomAttributes {...props} />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('when disabled', () => {
    beforeEach(() => {
      props = createTestProps({
        isDisabled: true,
      });
      wrapper = shallow(<CustomAttributes {...props} />);
    });
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should propagate `isDisabled` as `disabled`', () => {
      expect(wrapper.find(AttributeInput).at(0)).toHaveProp(
        'disabled',
        props.isDisabled
      );
    });
  });
});

describe('callbacks', () => {
  describe('when changing the attribute value', () => {
    let props;
    let wrapper;
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<CustomAttributes {...props} />);
      wrapper
        .find(AttributeInput)
        .at(0)
        .prop('onChangeValue')({ value: 'new value' });
    });
    it('should call the handleChange function', () => {
      expect(props.handleChange).toHaveBeenCalledTimes(1);
    });
    it('should call the handleChange function with the new value to change', () => {
      expect(props.handleChange).toHaveBeenCalledWith('test', 'new value');
    });
  });
});
