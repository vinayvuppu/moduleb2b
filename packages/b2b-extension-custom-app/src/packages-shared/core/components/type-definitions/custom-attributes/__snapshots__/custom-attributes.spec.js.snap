// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<div>
  <ul
    className="attribute-list"
  >
    <li
      className="attribute-list-item"
      key="test-0"
    >
      <AttributeInput
        attribute={
          Object {
            "name": "test",
            "value": "value",
          }
        }
        currencies={
          Array [
            "EUR",
            "USD",
          ]
        }
        definition={
          Object {
            "inputHint": undefined,
            "isRequired": undefined,
            "label": undefined,
            "name": "test",
            "type": Object {
              "name": "text",
            },
          }
        }
        disabled={false}
        expandSettings={Object {}}
        languages={
          Array [
            "en",
            "de",
          ]
        }
        numberFormat="en"
        onChangeValue={[Function]}
        selectedLanguage="en"
        updateSettings={[MockFunction]}
      />
    </li>
    <li
      className="attribute-list-item"
      key="boolean-1"
    >
      <AttributeInput
        attribute={
          Object {
            "name": "boolean",
            "value": false,
          }
        }
        currencies={
          Array [
            "EUR",
            "USD",
          ]
        }
        definition={
          Object {
            "inputHint": undefined,
            "isRequired": undefined,
            "label": undefined,
            "name": "boolean",
            "type": Object {
              "name": "boolean",
            },
          }
        }
        disabled={false}
        expandSettings={Object {}}
        languages={
          Array [
            "en",
            "de",
          ]
        }
        numberFormat="en"
        onChangeValue={[Function]}
        selectedLanguage="en"
        updateSettings={[MockFunction]}
      />
    </li>
    <li
      className="attribute-list-item"
      key="set-dates-2"
    >
      <AttributeInput
        attribute={
          Object {
            "name": "set-dates",
            "value": Array [
              "2017-01-27T09:40:38.179Z",
              "2017-05-27T09:40:38.179Z",
            ],
          }
        }
        currencies={
          Array [
            "EUR",
            "USD",
          ]
        }
        definition={
          Object {
            "inputHint": undefined,
            "isRequired": undefined,
            "label": undefined,
            "name": "set-dates",
            "type": Object {
              "name": "set",
            },
          }
        }
        disabled={false}
        expandSettings={Object {}}
        languages={
          Array [
            "en",
            "de",
          ]
        }
        numberFormat="en"
        onChangeValue={[Function]}
        selectedLanguage="en"
        updateSettings={[MockFunction]}
      />
    </li>
  </ul>
</div>
`;

exports[`rendering when disabled should match snapshot 1`] = `
<div>
  <ul
    className="attribute-list"
  >
    <li
      className="attribute-list-item"
      key="test-0"
    >
      <AttributeInput
        attribute={
          Object {
            "name": "test",
            "value": "value",
          }
        }
        currencies={
          Array [
            "EUR",
            "USD",
          ]
        }
        definition={
          Object {
            "inputHint": undefined,
            "isRequired": undefined,
            "label": undefined,
            "name": "test",
            "type": Object {
              "name": "text",
            },
          }
        }
        disabled={true}
        expandSettings={Object {}}
        languages={
          Array [
            "en",
            "de",
          ]
        }
        numberFormat="en"
        onChangeValue={[Function]}
        selectedLanguage="en"
        updateSettings={[MockFunction]}
      />
    </li>
    <li
      className="attribute-list-item"
      key="boolean-1"
    >
      <AttributeInput
        attribute={
          Object {
            "name": "boolean",
            "value": false,
          }
        }
        currencies={
          Array [
            "EUR",
            "USD",
          ]
        }
        definition={
          Object {
            "inputHint": undefined,
            "isRequired": undefined,
            "label": undefined,
            "name": "boolean",
            "type": Object {
              "name": "boolean",
            },
          }
        }
        disabled={true}
        expandSettings={Object {}}
        languages={
          Array [
            "en",
            "de",
          ]
        }
        numberFormat="en"
        onChangeValue={[Function]}
        selectedLanguage="en"
        updateSettings={[MockFunction]}
      />
    </li>
    <li
      className="attribute-list-item"
      key="set-dates-2"
    >
      <AttributeInput
        attribute={
          Object {
            "name": "set-dates",
            "value": Array [
              "2017-01-27T09:40:38.179Z",
              "2017-05-27T09:40:38.179Z",
            ],
          }
        }
        currencies={
          Array [
            "EUR",
            "USD",
          ]
        }
        definition={
          Object {
            "inputHint": undefined,
            "isRequired": undefined,
            "label": undefined,
            "name": "set-dates",
            "type": Object {
              "name": "set",
            },
          }
        }
        disabled={true}
        expandSettings={Object {}}
        languages={
          Array [
            "en",
            "de",
          ]
        }
        numberFormat="en"
        onChangeValue={[Function]}
        selectedLanguage="en"
        updateSettings={[MockFunction]}
      />
    </li>
  </ul>
</div>
`;
